Add `unstable_ir` feature flag that makes the ir pub

heesooy opened the pull request:
Fixes #1009

zhp46 commented:
This would be a fantastic addition to have

emilio reviewed (changes requested):
I guess this is reasonable to do, but I don't want to make any change to the ir a breaking change.

Let's call this `unstable_ir` and document it with no stability guarantees in docs.md (that is, if you use this, you need to pin the library, and are on your own).

With that, this looks reasonable to me, if it simplifies life downstream compared to a fork...

heesooy commented:
> I guess this is reasonable to do, but I don't want to make any change to the ir a breaking change.
> 
> Let's call this `unstable_ir` and document it with no stability guarantees in docs.md (that is, if you use this, you need to pin the library, and are on your own).
> 
> With that, this looks reasonable to me, if it simplifies life downstream compared to a fork...

@emilio -- done; renamed to `unstable_ir` and added a section in the docs. Let me know if you want me to move the docs to a different section, wasn't sure where would be best. Thanks for taking a look!
